addon/components/modal.js

```diff
@@ -88,6 +88,18 @@ export default Component.extend({
 	}),
 
 	/**
+	 * Has transitions property.
+	 *
+	 * @property hasTransitions
+	 * @type Boolean
+	 */
+	hasTransitions: computed(function() {
```

Comment:
If the transition property changes while the modal is opened, the method `_close` will receive the old value.
Use computed `volatile` or an es6 getter to prevent caching the property.